cardano-cli/test/Test/OptParse.hs

```diff
@@ -37,6 +41,20 @@ execCardanoCLI
   -- ^ Captured stdout
 execCardanoCLI = GHC.withFrozenCallStack $ H.execFlex "cardano-cli" "CARDANO_CLI"
 
+tryExecCardanoCLI
+  :: [String]
+  -- ^ Arguments to the CLI command
+  -> H.PropertyT IO (Either H.Failure String)
+  -- ^ Captured stdout, or error in case of failures
+tryExecCardanoCLI args =
+  GHC.withFrozenCallStack (H.execFlex "cardano-cli" "CARDANO_CLI") args
+    & H.unPropertyT
+    & H.unTest
+    & runExceptT
+    & lift
+    & H.TestT
+    & H.PropertyT
```

Comment:
What's the reason for introducing `tryExecCardanoCLI` instead of using `execCardanoCLI`?

Reply:
The desire to also test failing cases! Sadly, `try` or other exception handling mechanisms do not work inside of the `TestT` monad, so I had to extract and lift the error to be able to catch it and assert on it. 

There's maybe something more clever to do but I only had a day and a half to spend on all this so I'd rather "get it done".

Reply:
Reading the code again, it seems like you want to invert success/failure.

Reply:
I think that's okay for this PR.  Longer term we'd want to make the polarity more obvious.